### PR TITLE
Use new error constructorin task dao and service

### PR DESCRIPTION
**Pull request by Junoth**
(task:13602)

### CODE REVIEW

File: core/daov2/sqldb/task.go [hunk 1]
```diff
@@ -25,12 +25,11 @@ func (t Task) FindTaskByID(ct context.Context, taskID uint64) (entity.Task, *err
 		ReadOnly: true,
 	}
 	tx, err := t.transactionFactory.BeginTx(ct, &opt)
+
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go]:
```suggestion
```

File: core/daov2/sqldb/task.go [hunk 2]
```diff
@@ -25,12 +25,11 @@ func (t Task) FindTaskByID(ct context.Context, taskID uint64) (entity.Task, *err
...
 	if err != nil {
 		return entity.Task{}, err
 	}
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go:30]:
```suggestion
	}
	
```

File: core/daov2/sqldb/task.go [hunk 3]
```diff
@@ -39,10 +38,10 @@ func (t Task) FindTasksByTeamID(ct context.Context, teamID uint64) ([]entity.Tas
 		ReadOnly: true,
 	}
 	tx, err := t.transactionFactory.BeginTx(ct, &opt)
+
 	if err != nil {
 		return nil, err
 	}
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go:43]:
```suggestion
	}
	
```

File: core/daov2/sqldb/task.go [hunk 4]
```diff
@@ -39,10 +38,10 @@ func (t Task) FindTasksByTeamID(ct context.Context, teamID uint64) ([]entity.Tas
 		ReadOnly: true,
 	}
 	tx, err := t.transactionFactory.BeginTx(ct, &opt)
+
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go]:
```suggestion
```

File: core/daov2/sqldb/task.go [hunk 5]
```diff
@@ -97,13 +96,12 @@ func (t Task) FindTaskByIDWithTx(ct context.Context, tx *transaction.Transaction
...
-	if errors.Is(err, sql.ErrNoRows) {
-		internalErr := &errs.Error{
-			Code:    errs.NotFound,
-			Message: fmt.Sprintf("task not found: taskID=%v", taskID),
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return entity.Task{}, errs.NewError(errs.NotFound, fmt.Sprintf("task not found: taskID=%v", taskID))
+		} else {
+			return entity.Task{}, errs.NewError(errs.Unknown, err.Error())
 		}
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go]:
```suggestion
		}
		
		return entity.Task{}, errs.NewError(errs.Unknown, err.Error())
```

File: core/daov2/sqldb/task.go [hunk 6]
```diff
@@ -167,23 +159,13 @@ func (t Task) FindTasksByIDsWithTx(ct context.Context, tx *transaction.Transacti
...
-				Code:     errs.Unknown,
-				EmbedErr: err,
-			}
-
-			if internalErr == nil {
-				internalErr = newInternalErr
-			}
-
-			t.dataCollector.Logger.ErrorWithContext(ct, newInternalErr)
-			continue
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go:180]: Are we changing behavior from `continue getting the rest of the records` to `return on first error`?

File: core/daov2/sqldb/task.go [hunk 7]
```diff
@@ -223,22 +205,12 @@ func (t Task) FindTaskByCommentsThreadIDWithTx(ct context.Context, tx *transacti
...
-
 	if err != nil {
-		internalErr := &errs.Error{
-			Code:     errs.Unknown,
-			EmbedErr: err,
+		if errors.Is(err, sql.ErrNoRows) {
+			return entity.Task{}, errs.NewError(errs.NotFound, fmt.Sprintf("task not found: commentsThreadID=%v", commentThreadID))
+		} else {
+			return entity.Task{}, errs.NewError(errs.Unknown, err.Error())
 		}
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go]:
```suggestion
		}
		
		return entity.Task{}, errs.NewError(errs.Unknown, err.Error())
```

File: core/daov2/sqldb/task.go [hunk 8]
```diff
@@ -295,23 +261,13 @@ func (t Task) FindAllTasksWithTx(ct context.Context, tx *transaction.Transaction
...
-				EmbedErr: err,
-			}
-
-			if internalErr == nil {
-				internalErr = newInternalErr
-			}
-
-			t.dataCollector.Logger.ErrorWithContext(ct, newInternalErr)
-			continue
+			return nil, errs.NewError(errs.Unknown, err.Error())
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go:264]: Are we changing behavior from `continue getting the rest of the records` to `return on first error`?

File: core/daov2/sqldb/task.go [hunk 9]
```diff
@@ -368,17 +318,7 @@ func (t Task) FindTasksByTeamIDWithTx(ct context.Context, tx *transaction.Transa
...
```
Comment (naturetime10) [inline core/daov2/sqldb/task.go:321]: Are we changing behavior from `continue getting the rest of the records` to `return on first error`?

File: core/service/task.go [hunk 10]
```diff
@@ -477,17 +434,15 @@ func (t Task) tryIncreaseBandwidth(
...
-		participant, err := t.sprintParticipantDaoV2.FindParticipantWithTx(ct, tx, sprintID, taskOwnerUserID)
-		if err != nil {
-			if err.Code != errs.NotFound {
-				t.dataCollector.Logger.ErrorWithContext(ct, err)
-				return nil, err
+		participant, internalErr := t.sprintParticipantDaoV2.FindParticipantWithTx(ct, tx, sprintID, taskOwnerUserID)
+		if internalErr != nil {
+			if internalErr.Code != errs.NotFound {
+				return nil, internalErr
 			}
```
Comment (naturetime10) [inline core/service/task.go:446]:
```suggestion
			}
			
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: core/daov2/sqldb/task.go (hunk 6 above)
Comment (Junoth) [reply core/daov2/sqldb/task.go:180]: Yes, is there a use case to get rest of records aftering getting error in query?

Comment (naturetime10) [reply core/daov2/sqldb/task.go:180]: @Junoth Not at the moment. I am okay with behavior change

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**